src/table/slots.cc

```diff
@@ -10,31 +10,53 @@ bool startsWith(std::string const & a, std::string const b) {
     return a.compare(0, b.size(), b) == 0;
 }
 
+// helper class that resolves aliases for a slot's main measurement field, while
+// distiguishing between two cases where the field can't be found:
+//  - if the alias points to an invalid field, we should throw an exception.
+//  - if the alias simply isn't defined, we should just reset the slot with invalid keys
+class MeasFieldNameGetter {
```

Comment:
It might help readability to be more specific about exactly where the throwing, etc, is supposed to happen. Being unfamiliar with the code, I was confused to note that you seemed to be catching and handling exceptions below rather than raising them. Took me a moment to realise where the exception you're referring to here was coming from.